cypress/e2e/Admin/results/TableByQuestion.cy.js

```diff
@@ -0,0 +1,203 @@
+
+describe('Table by Question', () => {
+  /**
+   * Test the table by question view for a few question and some user answers
+   */
+  it('Table by Question correctly display data', () => {
+    cy.setUpApi({
+      database: {
+        appSettings: APP_SETTINGS2,
+        appData: APP_DATA,
+      },
+      appContext: {
+        permission: PERMISSION_LEVELS.ADMIN,
+        context: CONTEXTS.BUILDER,
+      },
+    });
+    cy.visit('/');
+
+    // navigate to the table by question
+    cy.get(dataCyWrapper(NAVIGATION_RESULT_BUTTON_CY)).click();
+
+    // Test that each table are correctly displayed
+    APP_SETTINGS2.filter((s) => s.name === APP_SETTING_NAMES.QUESTION).forEach(
+      (s, idx) => {
+        cy.get(dataCyWrapper(buildTableByQuestionCy(s.data.question))).should(
+          'have.text',
+          APP_SETTINGS2[idx].data.question
+        );
+
+        // test header
+        testTableHeader(s.data.question, 'sorted ascending');
+        // test content
+        testTableContent(s.data.question, true);
+
+        // sort descending
+        cy.get(
+          dataCyWrapper(buildTableByQuestionUserHeader(s.data.question))
+        ).click();
+
+        // test header
+        testTableHeader(s.data.question, 'sorted descending');
+        // test content
+        testTableContent(s.data.question, false);
+      }
+    );
+  });
+});
+
+/**
+ * Helper function to test that the header of the table is correct
+ *
+ * @param {string} qTitle The title of the question for which we currently are displaying data
+ * @param {string} ascending Suffix to add to User whether it is sorted ascending or descending
+ */
+const testTableHeader = (qTitle, ascending) => {
+  cy.get(dataCyWrapper(buildTableByQuestionUserHeader(qTitle))).should(
+    'have.text',
+    `User${ascending}`
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionAnswerHeader(qTitle))).should(
+    'have.text',
+    'Answer'
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionDateHeader(qTitle))).should(
+    'have.text',
+    'Date'
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionCorrectHeader(qTitle))).should(
+    'have.text',
+    'Correct'
+  );
+};
+
+/**
+ * Helper function to test that the content of the table is correct
+ *
+ * @param {string} qTitle The title of the question for which we currently are displaying data
+ * @param {boolean} ascending Whether the current sorting order is ascending or descending
+ */
+const testTableContent = (qTitle, ascending) => {
+  const n = ascending ? 0 : 2;
+  cy.get(dataCyWrapper(buildTableByQuestionTableBodyCy(qTitle)))
+    .children('tr')
+    .each((tr, idxTr) => {
+      // test for the header of the row
+      cy.wrap(tr)
+        .children('th')
+        .first()
+        .should('have.text', RESPONSES[qTitle][Math.abs(n - idxTr)].userId);
+
+      // test for the content of the row
+      cy.wrap(tr)
+        .children('td')
+        .each((td, idxTd) => {
+          if (idxTd === 2) {
```

Comment:
This means if we change the order/add a column this will fail right? Why not test over the column name/selector?

Reply:
Yes you are right, it constraint the table too much to check it like that. I modified it to retrieve the column entry given a selector to check its content. It can now be reordered, or a new column can easily be added without breaking the test.